### DISCUSSION TITLE
Scheduled task wordt update applications..

### DISCUSSION THREAD

**Discussion by dvintial** [Romanitho/Winget-AutoUpdate#547] [Q&A]
Dear fellow updaters,

I really enjoy this auto-update application. Tested it stand-alone and it works perfectly.
However when deploy it through intune, the scheduled tasks made wont update any application.

There seem to be no application which matches selection criteria or everything seems to be updated..(WAU logs) And when i check winget logs it gives an error that says no file which already exists can be created..

When i execute the powershell script by hand it works fine! Its all so strange and i cannot find the root cause of this error..
Is there someone outta here who has had the same issue or can get me on track solving this issue?

Thank you for reading.

Kind regards

**Comment by Romanitho** [comment]
Hello. Maybe winget is not up to date?

**Comment by pascalpoulincssbe** [comment]
Same  probleme. Wau tell that it have an application to update, but at the end, no application are updated. Some times , it works, some times, it don't. Here is the log files
[updates.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14933662/updates.log)

**Reply by pascalpoulincssbe** [reply]
Winget = 1.7.10861

**Comment by AndrewDemski-ad-gmail-com** [comment]
The "No available upgrade found." message does not come from WAU.
However, this line exists in winget-cli.
https://github.com/microsoft/winget-cli/blob/91ee267ccd1bb6d9f976153e195572a5ee9746d8/src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw#L703

It seems that for some reason whenever WAU calls the winget upgrade, it starts from re-downloading that sqlLite file with db of winget repository.
Then it fails.

Have you considered monitoring the winget.exe process to see if it is terminating after each "upgrade" call without problems?
I could speculate that AV is keeping the file hostage,

Don't shoot the messenger :)

**Reply by pascalpoulincssbe** [reply]
Thanks a lot for your answer. I'm not sure how to monitor the AV. We use Microsoft Defender. If i check the winget logs in C:\Windows\Temp\WinGet\defaultState, nothing seems to terminate the winget process. I attach all the winget logs. Can you help me with this?
[WinGet-2024-04-11-08-03-33.636.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946239/WinGet-2024-04-11-08-03-33.636.log)
[WinGet-2024-04-11-08-03-32.415.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946240/WinGet-2024-04-11-08-03-32.415.log)
[WinGet-2024-04-11-08-03-27.478.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946241/WinGet-2024-04-11-08-03-27.478.log)
[WinGet-2024-04-11-08-03-18.247.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946242/WinGet-2024-04-11-08-03-18.247.log)
[WinGet-2024-04-11-08-03-16.917.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946243/WinGet-2024-04-11-08-03-16.917.log)
[WinGet-2024-04-11-08-03-11.898.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946244/WinGet-2024-04-11-08-03-11.898.log)
[WinGet-2024-04-11-08-03-05.150.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946245/WinGet-2024-04-11-08-03-05.150.log)
[WinGet-2024-04-11-08-03-04.239.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946246/WinGet-2024-04-11-08-03-04.239.log)
[WinGet-2024-04-11-08-02-58.682.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946247/WinGet-2024-04-11-08-02-58.682.log)
[WinGet-2024-04-11-08-02-55.650.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946248/WinGet-2024-04-11-08-02-55.650.log)
[WinGet-2024-04-11-08-02-54.859.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946249/WinGet-2024-04-11-08-02-54.859.log)
[WinGet-2024-04-11-08-03-47.236.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946250/WinGet-2024-04-11-08-03-47.236.log)
[WinGet-2024-04-11-08-03-45.976.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946251/WinGet-2024-04-11-08-03-45.976.log)
[WinGet-2024-04-11-08-03-41.065.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14946252/WinGet-2024-04-11-08-03-41.065.log)

**Comment by AndrewDemski-ad-gmail-com** [comment]
According to [updates.log](https://github.com/Romanitho/Winget-AutoUpdate/files/14933662/updates.log) Winget encountered errors around those timestamps
- 2024-04-10 10:06:32
- 2024-04-10 10:06:45
- 2024-04-10 10:06:59
- 2024-04-10 10:09:50
I do not see those fraction logs here.

Although I see several instances of 
`[CLI ] Terminating context: 0x8a15002b at C:\__w\1\s\external\pkg\src\AppInstallerCLICore\Workflows\UpdateFlow.cpp:bc`

which can be traced back to .. again.. winget itself.
https://github.com/microsoft/winget-cli/blob/91ee267ccd1bb6d9f976153e195572a5ee9746d8/src/AppInstallerCommonCore/AppInstallerTelemetry.cpp#L325
Which is called by logic in ExecutionContext.cpp
https://github.com/microsoft/winget-cli/blob/91ee267ccd1bb6d9f976153e195572a5ee9746d8/src/AppInstallerCLICore/ExecutionContext.cpp#L408
in a Context::Terminate function.

That code leads nowhere, because in [winget error codes table](https://github.com/microsoft/winget-cli/blob/master/doc/windows/package-manager/winget/returnCodes.md) it states the obvious:

Hex | Decimal | Symbol | Description
-- | -- | -- | --
0x8A15002B | -1978335189 | APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE | No applicable update found

I am still convinced that it is a "winget-having-problem-with-winget" kind of issue.

One of solutions I found while googling for clues was this.
Would not hurt checking if that solves the issue.
https://github.com/microsoft/winget-cli/issues/998#issuecomment-909680518
